### DISCUSSION TITLE
Data Validation / QA

### DISCUSSION THREAD

**Discussion by n0rdlicht** [cividi/handbook#6] [Q&A]
Discussion thread for https://github.com/cividi/handbook/wiki/Data-Validation

**Comment by bkarolina** [comment]
These are things I would need to write down first: 
* Process of Swisscom data --> can be reused from 014-Wohlen project (process files to database)
* Prefect workflow --> I haven't docs written yet, as it will be open sourced and there should be as well part "how to contribute"

**Comment by n0rdlicht** [comment]
Not sure where to put workflow tips and tricks like these yet:

### Leverage git as a monitoring tool
What: Monitor changes to output/process files to avoid unintentional side effects of code with version control while working
How:
- version (e.g. with git) all outputs and relevant process artefacts as plain text files (e.g. CSV or JSON)
- if possible version changes to code/workflows/... generating the files and source files in the same commit
- keep the same name for the same output/artefacts and rely on git to recover old versions rather than renaming / copying
- come up with the data structure as early as possible to be able to track changes to the output easily
- if exports to binary/propriatery file formats are needed do so as a very last export step from CSV/JSON files
